sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -337,26 +329,29 @@ export class KeyVaultAccessControlClient {
   private async *listRoleDefinitionsPage(
     roleScope: KeyVaultRoleScope,
     continuationState: ListRoleDefinitionsPageSettings,
-    options?: ListRoleDefinitionsOptions
+    options: ListRoleDefinitionsOptions = {}
   ): AsyncIterableIterator<KeyVaultRoleDefinition[]> {
     if (!continuationState.continuationToken) {
       const optionsComplete: RoleAssignmentsListForScopeOptionalParams = options || {};
-      const currentSetResponse = await this.client.roleDefinitions.list(
-        this.vaultUrl,
-        roleScope,
-        optionsComplete
+      const currentSetResponse = await withTrace(
+        "listRoleDefinitions",
+        optionsComplete,
+        (updatedOptions) =>
+          this.client.roleDefinitions.list(this.vaultUrl, roleScope, updatedOptions)
       );
       continuationState.continuationToken = currentSetResponse.nextLink;
       if (currentSetResponse.value) {
         yield currentSetResponse.value.map(mappings.roleDefinition.generatedToPublic, this);
       }
     }
     while (continuationState.continuationToken) {
-      const currentSetResponse = await this.client.roleDefinitions.listNext(
-        this.vaultUrl,
-        roleScope,
-        continuationState.continuationToken,
-        options
+      const currentSetResponse = await withTrace("listRoleDefinitions", options, (updatedOptions) =>
```

Comment:
Yeah, it's a good observation. @richardpark-msft and I had a chat about this as well when we were pairing. My rationale for keeping the two named the same is that from the user's perspective the difference between the first fetch and the Nth fetch isn't meaningful, it's just an implementation detail. From their perspective they are (asynchronously) iterating over items and every once in a while a call is made to fetch another page. 

We didn't even name it `listRoleDefinitionsPage` because that's another implementation detail that this is where the remote calls are happening. 

But I can definitely see an argument for naming them differently if we think that could help with tracing. One concern is how will the trace look if there is only one page - you might see "listRoleDefinitionsInitial" followed by nothing which might also be confusing / misleading. What do you think?

Reply:
I think that rationale is fine! I don't really have enough tracing context to know what is best.